include/utils/tree_node.h

```diff
@@ -209,6 +218,17 @@ class TreeNode
   const unsigned int tgt_bin_size;
 
   /**
+   * The current refinement level of this TreeNode.
+   */
+  unsigned int refinement_level;
+
+  /**
+   * This specifies the refinement level beyond which we will
+   * scale up the target bin size in child TreeNodes.
+   */
```

Comment:
Comment on the default here?  It is a (deserved!) behavior change.  We should also wait and see whether others prefer more or less than 10 levels, though that certainly sounds reasonable to me.